### DISCUSSION TITLE
Where's the task source?

### DISCUSSION THREAD

**Discussion by buggtb** [microsoft/farmvibes-ai#92] [Q&A] [answered]
Hello folks

Curious to know about the tasks and stuff that land in the docker images and in turn become available to workflows.

I poked around a while ago and found them inside a docker container, whilst I can poke around in the docker image to find these things, is the published online somewhere? Because these images don't come from this repo and there's no obvious place in the docs pointing to where they do originate from.

**Comment by renatolfc** [comment]
Hi @buggtb,

As of now, the source of those components haven’t been open-sourced yet.

They are currently built by an internal pipeline that pushes them to the public MCR repo the cluster pulls images from.

To open source the ops (tasks), we’d probably need to also open source the tooling and the associated worker code.

I’ll check internally what are our plans related to that.

**Reply by buggtb** [reply]
Thanks @renatolfc 

For those of us wanting to use the platform, it feels a little weird having an open source framework, pulling a bunch of closed source stuff and the former covered by a MIT license and the latter not obviously covered by any license. So by adopting the framework there's certainly a concern that either Microsoft decide to remove the closed source bits and or they decide that you can't use the closed source bits in a commercial type setup, which would certainly cause adoption concerns.

**Reply by renatolfc** [reply, marked as answer]
Hello @buggtb, you might want to take a look at [the ops directory under the dev branch](https://github.com/microsoft/farmvibes-ai/tree/dev/ops).

**Comment by RiyazPishori** [comment]
Hi buggtb, 
We are still exploring how best to expose it. In the meantime, we can work with you offline to see how we unblock your commercial use case. 
Would love to learn about your use case and workflows that you are interested in as well. 
Tom, you can reach me at riyazp at microsoft dot com
Let us know, 
Thanks,
Riyaz
